### PR TITLE
Update the docs with terraform-plugin-docs

### PR DESCRIPTION
**Pull request by datianshi**
This folder's content is not updated for over a year
This doc is generated by https://github.com/hashicorp/terraform-plugin-docs. to reflect the current provider states. 
This will rendered to registry.terraform.io once we published it. Note: This is more of skeleton and we should continue iterate and document. 

Also, copy the README.md content to index.md

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -60,34 +65,158 @@ resource "ocm_group_membership" "my_admin" {
...
+          role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.operator_role_prefix}-openshift-machine-api-aws-cloud-credentials",
+        },
+      ]
+      instance_iam_roles = {
+        master_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-ControlPlane-Role",
+        worker_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Worker-Role"
+      },    
+  }
+}
+resource "ocm_cluster" "rosa_cluster" {
```
Comment (bardielle) [inline docs/index.md:125]: Please use the dedicated resource ocm_cluster_rosa_classic
ocm_cluster doesn't support STS anymore... 
Update it like in that example -https://github.com/terraform-redhat/terraform-provider-ocm/blob/main/examples/create_rosa_cluster/create_rosa_sts_cluster/cluster/main.tf#L50

File: docs/index.md [hunk 2]
```diff
@@ -60,34 +65,158 @@ resource "ocm_group_membership" "my_admin" {
...
+* STS
+
+```
+data "aws_caller_identity" "current" {}
+
+locals {
+  sts_roles = {
+      role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Installer-Role",
+      support_role_arn = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/ManagedOpenShift-Support-Role",
+      operator_iam_roles = [
```
Comment (bardielle) [inline docs/index.md:87]: You only need the operator_role_prefix
Please see the example:  https://github.com/terraform-redhat/terraform-provider-ocm/blob/main/examples/create_rosa_cluster/create_rosa_sts_cluster/cluster/main.tf#L36

File: docs/index.md [hunk 3]
```diff
@@ -60,34 +65,158 @@ resource "ocm_group_membership" "my_admin" {
...
+    https_proxy = var.proxy
+  }
+  properties = {
+    rosa_creator_arn = data.aws_caller_identity.current.arn
+  }
+  wait = false
+  sts = local.sts_roles
+}
+
+module sts_roles {
```
Comment (bardielle) [inline docs/index.md:147]: It was decided to split the operator IAM roles creation from the cluster:

- In the destroy process the operator IAM roles and the OIDC provider would be deleted before the cluster would finish the uninstall flow
- In order to support BYO operator IAM roles

File: docs/index.md [hunk 4]
```diff
@@ -60,34 +65,158 @@ resource "ocm_group_membership" "my_admin" {
...
+[registry](https://registry.terraform.io/providers/rh-mobb/ocm/latest/docs).
 
-- **client_secret** (String, Sensitive) OpenID client secret.
+## Examples
 
-- **insecure** (Boolean) When set to `true` enables insecure communication
-  with the server. This disables verification of TLS certificates and host names
-  and it isn't recommended for production environments. The default value is
-  `false`.
+Check the [examples](examples) directory for complete examples.
```
Comment (bardielle) [inline docs/index.md:165]:
```suggestion
Check the [examples](https://github.com/terraform-redhat/terraform-provider-ocm/blob/main/examples/create_rosa_cluster/create_rosa_sts_cluster) directory for complete examples.
```

File: docs/index.md [hunk 5]
```diff
@@ -60,34 +65,158 @@ resource "ocm_group_membership" "my_admin" {
...
+}
+```
+
 
-### Optional
+## Documentation
 
-- **client_id** (String) OpenID client identifier.
+The reference documentation of the provider is available in the Terraform
+[registry](https://registry.terraform.io/providers/rh-mobb/ocm/latest/docs).
```
Comment (bardielle) [inline docs/index.md:161]:
```suggestion
[registry](https://registry.terraform.io/providers/terraform-redhat/ocm/latest/docs).
```

File: docs/index.md [hunk 6]
```diff
@@ -60,34 +65,158 @@ resource "ocm_group_membership" "my_admin" {
...
+  }
+  properties = {
+    rosa_creator_arn = data.aws_caller_identity.current.arn
+  }
+  wait = false
+  sts = local.sts_roles
+}
+
+module sts_roles {
+    source  = "rh-mobb/rosa-sts-roles/aws"
```
Comment (bardielle) [inline docs/index.md:148]: This module doesn't support a dynamic operator role names... 
I write it from scratch - here is the module https://github.com/terraform-redhat/terraform-provider-ocm/tree/main/modules/aws_roles and it contains one template for all the operator roles (in order to support dynamic names) https://github.com/terraform-redhat/terraform-provider-ocm/blob/main/modules/aws_roles/operator_roles/operator_role_resource.tf 


I'm working on moving it out into a new repository 

Please change it to be like in that example - https://github.com/terraform-redhat/terraform-provider-ocm/blob/main/examples/create_rosa_cluster/create_rosa_sts_cluster/operator_roles_and_oidc/main.tf#L41
`    source  = "git::https://github.com/terraform-redhat/terraform-provider-ocm.git//modules/aws_roles"
`
And when I will move it out into a new repository I'll change it here also

**Comment by bardielle** [comment]
@datianshi please use the `tfplugindocs` tool as mentioned [here](https://developer.hashicorp.com/terraform/tutorials/providers/provider-release-publish#document-your-provider), in order to automatically generate documentation in the format required by the Terraform Registry

**Comment by datianshi** [comment]
@bardielle  This is using the tfplugindocs tool.  Since you created another PR, to avoid rebase and confusion. I will just delete this one.

**Pull request closed without merging**